### PR TITLE
soc: esp32c6: add Wi-Fi support

### PR DESCRIPTION
**Pull request by sylvioalves**
Enables Wi-Fi support.

### CODE REVIEW

File: drivers/clock_control/clock_control_esp32.c [hunk 1]
```diff
@@ -87,6 +87,20 @@ static bool reset_reason_is_cpu_reset(void)
...
+	modem_clock_lpclk_src_t modem_lpclk_src =
+		(modem_clock_lpclk_src_t)((rtc_slow_clk_src == SOC_RTC_SLOW_CLK_SRC_RC_SLOW)
+						  ? MODEM_CLOCK_LPCLK_SRC_RC_SLOW
+					  : (rtc_slow_clk_src == SOC_RTC_SLOW_CLK_SRC_XTAL32K)
+						  ? MODEM_CLOCK_LPCLK_SRC_XTAL32K
+					  : (rtc_slow_clk_src == SOC_RTC_SLOW_CLK_SRC_RC32K)
+						  ? MODEM_CLOCK_LPCLK_SRC_RC32K
+					  : (rtc_slow_clk_src == SOC_RTC_SLOW_CLK_SRC_OSC_SLOW)
+						  ? MODEM_CLOCK_LPCLK_SRC_EXT32K
+						  : SOC_RTC_SLOW_CLK_SRC_RC_SLOW);
```
Comment (sylvioalves) [inline drivers/clock_control/clock_control_esp32.c:100]: Used calng-format here.

**Comment by raffarost** [comment]
LGTM

**Comment by wmrsouza** [comment]
LGTM

**Review by marekmatej — DISMISSED** [review]

**Comment by kartben** [comment]
looks like now would be a good time to update the "Supported Features" table for the ESP32-C6 DevkitC (don't think there are other C6 boards... yet) to not forget..?

**Review by kartben — DISMISSED** [review]
Cool! Looks like I will need to find some time to finish #74697 :)

**Review by uLipe — DISMISSED** [review]

**Review by wmrsouza — DISMISSED** [review]

**Review by raffarost — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by raffarost — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**